packages/ptokens-swap/src/ptokens-swap.ts

```diff
@@ -137,17 +157,29 @@ export class pTokensSwap {
               .on('txConfirmed', (_swapResult: SwapResult) => {
                 promi.emit('inputTxConfirmed', _swapResult)
               })
-            const outputTx = await this.monitorOutputTransactions(swapResult.operationId)
+            const interimTxHash = await this.monitorInterimTransactions(swapResult.operationId)
```

Comment:
I would change monitorCrossChainOperations implementation in a way it returns the log instead of tx hash.

In this way you can pass the log directly into the already existing getOperationIdFromLog

Reply:
This would mean to import `Log` type on ptoken-swap. It is not manadatory here but SwapResult should return the Log as `any` otherwise